DAGE-25: v1 working demo

nicolo-rinaldi opened the pull request:
Full pipeline in the file `dataset_generator.py`. Number of total queries fixed. Implemented the score of the full cartesian product between retrieved documents and queries. 

[ticket](https://sease.atlassian.net/browse/DAGE-25)

nicolo-rinaldi committed:
[DAGE-15] changed config to have less value and started dataset_generator.py full pipeline

nicolo-rinaldi committed:
[DAGE-15] updated dataset_generator.py

nicolo-rinaldi committed:
[DAGE-15] full (provisory) pipeline coded in file dataset_generator.py

nicolo-rinaldi committed:
[DAGE-15] build of integration tests

Changes:
- added a test for utils.py functions
- started building the test_integration.py file to mock and test a full pipeline, to see if all the blocks interact with each other as expected

nicolo-rinaldi committed:
[DAGE-15] Added first version of main script and resolved the conflicts

nicolo-rinaldi committed:
[DAGE-25] First working demo

- Modified config.yaml to be launched with the demo
- updated dataset_generator.py to be a full pipeline. Right now, it has errors (needed upgrade in the data_store.py file)

nicolo-rinaldi committed:
[DAGE-25] First working demo

- Modified config.yaml to be launched with the demo
- updated dataset_generator.py to be a full pipeline. Right now, it has errors (needed upgrade in the data_store.py file)

nicolo-rinaldi committed:
[DAGE-25] minor changes to dataset_generator.py

nicolo-rinaldi committed:
[DAGE-25] Updated code according to data_store.py new features

nicolo-rinaldi committed:
[DAGE-25] added empty lines

dantuzi commented on rre-dataset-generator/dataset_generator.py:
```diff
@@ -16,12 +21,88 @@
+        if flag:
+            break
+
+    # retrieval of the document we need to store for user queries
+    for query_text in user_queries:
+        docs_eval = search_engine.fetch_for_evaluation(keyword=query_text,
+                                                       query_template=config.query_template,
+                                                       doc_fields=config.doc_fields)
+        for doc in docs_eval:
+            if not data_store.has_document(doc.id):
+                data_store.add_document(doc.id, doc)
+            data_store.add_query(query_text, doc.id)
+
+    # retrieval of the document we need to store for generated queries
+    for query_rating_context in data_store.get_queries():
+        docs_eval = search_engine.fetch_for_evaluation(keyword=query_rating_context.get_query(),
+                                                       query_template=config.query_template,
+                                                       doc_fields=config.doc_fields)
+        for doc in docs_eval:
+            if not data_store.has_document(doc.id):
+                data_store.add_document(doc.id, doc)
+            data_store.add_query(query_rating_context.get_query(), doc.id)
+
+    log.debug(f"Number of documents evaluated: {len(docs_to_generate_queries)}")
+    for query_rating_context in data_store.get_queries():
+        for doc in data_store.get_documents():
+            data_store.add_query(query_rating_context.get_query(), doc.id)
```
This is out of the scope of this ticket

nicolo-rinaldi replied:
removed

dantuzi commented on rre-dataset-generator/dataset_generator.py:
```diff
@@ -16,12 +21,88 @@
+                                                                        doc_number=config.doc_number,
+                                                                        doc_fields=config.doc_fields)
+    log.debug(f"Number of documents retrieved for generation: {len(docs_to_generate_queries)}")
+    llm = build_openai(LLMConfig.load(config.llm_configuration_file))
```
How about if the user adds a parameter to work with Gemini? Do we need to change the code?

nicolo-rinaldi replied:
Yes, we need add an `if` statement

dantuzi replied:
Let's use the factory pattern instead. Something like
```
LLMService = Factory.build_llm_service(config)
```

nicolo-rinaldi replied:
done

dantuzi commented on rre-dataset-generator/dataset_generator.py:
```diff
@@ -16,12 +21,88 @@
+    else:
+        error_msg = "Other format than 'quepid' are not yet supported"
+        log.error(error_msg)
+        raise NotImplementedError(error_msg)
```
I would have used the factory pattern: https://www.tutorialspoint.com/design_pattern/factory_pattern.htm

nicolo-rinaldi replied:
Ok, we need to update the writer with a writer factory then

nicolo-rinaldi replied:
I implemented this

nicolo-rinaldi committed:
[DAGE-25] Modified main file

- modified dataset_generator.py file according to PR
- added factory design patterns for both search engines and writers

nicolo-rinaldi committed:
[DAGE-25] minor change

nicolo-rinaldi committed:
[DAGE-25] Moved factory for llm and adapted code in dataset_generator.py
